test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -3232,3 +3235,82 @@ UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols_err_sender)
 UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(test_ucp_sockaddr_protocols_err_sender,
                                         rc_no_ud, "rc_mlx5,rc_verbs")
+
+
+class test_ucp_sockaddr_iface_activate : public test_ucp_sockaddr {
+public:
+    static void
+    get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        get_test_variants_mt(variants, UCP_FEATURE_TAG,
+                             CONN_REQ_TAG | TEST_MODIFIER_CM_USE_ALL_DEVICES,
+                             "tag");
+    }
+
+    virtual int is_skip_interface(const struct ifaddrs *ifa) const
+    {
+        return test_ucp_sockaddr::is_skip_interface(ifa) ||
+               ucs::is_rdmacm_netdev(ifa->ifa_name);
+    }
+
+    template <typename Func>
+    void wait_progress(Func func)
+    {
+        ucs_time_t deadline = ucs::get_deadline();
+        scoped_log_handler slh(detect_error_logger);
+        while (!func() && (ucs_get_time() < deadline)) {
+            progress();
+        };
+    }
+
+    ucp_worker_iface_t *get_test_iface(entity &e) const
+    {
+        ucp_worker_h worker = e.worker();
+        ucp_context_h ctx   = worker->context;
+
+        for (unsigned i = 0; i < worker->num_ifaces; ++i) {
+            ucp_worker_iface_t *wiface = worker->ifaces[i];
+            const char *name = ctx->tl_rscs[wiface->rsc_index].tl_rsc.dev_name;
+            if (m_test_addr.netdev_name() == name) {
+                return wiface;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void client_connect_disconnect(size_t sender_index)
+    {
+        entity &sender = e(sender_index);
+
+        /* Check state before connection establishment */
+        ucp_worker_iface_t *sender_iface = get_test_iface(sender);
+        ucp_worker_iface_t *server_iface = get_test_iface(receiver());
+        EXPECT_EQ(0, ucp_worker_iface_is_activated(sender_iface));
+        EXPECT_EQ(0, ucp_worker_iface_is_activated(server_iface));
+
+        /* Connect sender and receiver */
+        client_ep_connect_basic(get_ep_params(), 0, false, sender_index);
+        wait_progress([&]{return ucp_worker_iface_is_activated(server_iface);});
+
+        /* Check state after connection establishment */
+        EXPECT_EQ(sender_index + 1, receiver().get_num_eps());
+        EXPECT_EQ(0, sender.get_err_num());
+        EXPECT_EQ(1, ucp_worker_iface_is_activated(sender_iface));
+        EXPECT_EQ(1, ucp_worker_iface_is_activated(server_iface));
+
+        one_sided_disconnect(sender);
+        EXPECT_EQ(0, ucp_worker_iface_is_activated(server_iface));
+        receiver().reset_err_num();
+    }
+};
+
+UCS_TEST_P(test_ucp_sockaddr_iface_activate, iface_activate_count)
+{
+    create_entity(true);
+    listen(cb_type());
+    client_connect_disconnect(0);
+    client_connect_disconnect(1);
```

Comment:
why need to use different index (1) here, if index 0 is already disconnected?
maybe reuse index 0 and then could avoid adding sender_index parameter to different function